addons/survey/static/src/js/survey_form.js

```diff
@@ -253,17 +254,30 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
-    _onSubmit: function (event) {
+    _onSubmit(event) {
         event.preventDefault();
         const options = {};
         const target = event.currentTarget;
         if (target.value === 'previous') {
             options.previousPageId = parseInt(target.dataset['previousPageId']);
         } else if (target.value === 'next_skipped') {
             options.nextSkipped = true;
-        } else if (target.value === 'finish') {
+        } else if (target.value === 'finish' && target.innerText == 'Submit') {
             options.isFinish = true;
-        }
+        } else if (target.value === 'finish') {
```

Comment:
How come there is a case where value is `finish` and text isn't `submit`? Can't/shouldn't we fix this instead?

(We can then decide to add the confirmation dialog anyway if we don't think it's clear enough, but then it will be in all cases).

Reply:
Hello @flch-odoo 

> How come there is a case where value is finish and text isn't submit? 

It's the arrow button on the button right corner :sweat_smile: (navigation arrow)
The whole point of the task is to warn users about this one :smile: 

> Can't/shouldn't we fix this instead?

Haven't thought about it this way but sure, we can add 'Submit' text if its the last question in the survey. Let's see what @jija2  has to say about this.

Thanks!